src/axolotl/utils/config/models/input/v0_4_1/__init__.py

```diff
@@ -1548,3 +1550,21 @@ def check_multigpu_unsloth(cls, data):
                     "unsloth_lora_mlp, unsloth_lora_qkv, and unsloth_lora_o are not compatible with multi-GPU training."
                 )
         return data
+
+    @model_validator(mode="before")
+    @classmethod
+    def check_adopt_torch_version(cls, data):
+        if (data.get("optimizer") is not None) and ("adopt" in data.get("optimizer")):
+            env_capabilities = data.get("env_capabilities")
```

Comment:
```suggestion
            env_capabilities = data.get("env_capabilities", {})
```

this should help with the case where env_capabilities is None